karapace/backup/api.py

```diff
@@ -303,15 +293,34 @@ def _write_partition(
     )
 
 
+def _handle_restore_topic_legacy(
+    instruction: RestoreTopicLegacy,
+    config: Config,
+) -> None:
+    if config["topic_name"] != instruction.name:
+        LOG.warning(
+            "Not creating topic, because the name %r from the config and the name %r from the CLI differ.",
+            config["topic_name"],
+            instruction.name,
+        )
+        return
```

Comment:
This is slightly more complex than that because:
- `replication_factor` for V1/V2 is taken from configuration and it's not hard coded, so in order to correctly create a `RestoreTopic` we would need to edit the `BaseBackupReader.read` interface to accept also the config as parameter
- `_handle_restore_topic` also needs to distinguish between V1/V2 and V3 behaviour when dealing with a topic already existing.

To be able to have a single `RestoreTopic` type of instruction probably the cleanest way would be to insert the version inside the `RestoreTopic` dataclass, but again we currently cannot distinguish between V1 and V2 (even though it's not really meaningful in this case).

Overall I think that having different instructions and different handlers is more clean than having a single instruction with more attributes and a single handler that handles different cases with conditional logic. All of this is IMHO of course

Reply:
Ah, this makes sense 👍